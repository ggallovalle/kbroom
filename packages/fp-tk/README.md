# fp-tk

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test fp-tk` to execute the unit tests via [Jest](https://jestjs.io).
