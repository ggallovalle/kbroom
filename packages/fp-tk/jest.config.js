module.exports = {
  displayName: "fp-tk",
  preset: "../../jest.preset.js",
 "setupFilesAfterEnv": ["jest-extended"],
  globals: {
    "ts-jest": {
      tsconfig: "<rootDir>/tsconfig.spec.json",
    },
  },
  testEnvironment: "node",
  transform: {
    "^.+\\.[tj]sx?$": "ts-jest",
  },
  moduleFileExtensions: ["ts", "tsx", "js", "jsx"],
  coverageDirectory: "../../coverage/packages/fp-tk",
};
