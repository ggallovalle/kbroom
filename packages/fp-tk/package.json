{
  "name": "fp-tk",
  "version": "2.0.0",
  "license": "MIT",
  "peerDependencies": {
    "fp-ts": "2.10.5",
    "io-ts": "2.2.16",
    "monocle-ts": "2.3.10",
    "newtype-ts": "0.3.4",
    "fp-ts-rxjs": "0.6.15"
  }
}
