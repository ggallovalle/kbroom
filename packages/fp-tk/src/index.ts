export * from "./lib/fp-ts";
export * from "./lib/fp-ts-rxjs";
export * from "./lib/io-ts";
export * from "./lib/monocle-ts";
import * as nt from "./lib/newtype-ts";
export { thrower } from "./lib/extensions/Error";
export * as str from "./lib/extensions/String";

export { nt };
