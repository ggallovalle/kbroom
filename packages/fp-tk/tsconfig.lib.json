{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "CommonJS",
    "target": "ES2015",
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "exclude": ["**/*.spec.ts"],
  "include": ["**/*.ts"]
}
