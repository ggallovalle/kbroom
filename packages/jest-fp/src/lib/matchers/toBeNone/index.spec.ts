import { option } from "fp-tk";
import matcher from "./";

expect.extend(matcher);

describe(".toBeNone", () => {
  it("should pass when given none", () => {
    // @ts-expect-error toBeNone isn't added to typescript types
    expect(option.none).toBeNone();
    //- @ts-expect-error toBeNone isn't added to typescript types
    // expect("hola").toBeNone();
  });

  it.skip("should failed when given something that isn't none", () => {
    // @ts-expect-error toBeNone isn't added to typescript types
    expect(() => expect(true).toBeNone()).toThrowErrorMatchingSnapshot();
  });
});
// describe(".toBeTrue", () => {
//   test("passes when given true", () => {
//     expect(true).toBeTrue();
//   });

//   test("fails when not given true", () => {
//     expect(() => expect(false).toBeTrue()).toThrowErrorMatchingSnapshot();
//   });
// });

// describe(".not.toBeTrue", () => {
//   each([
//     [false],
//     [""],
//     [0],
//     [{}],
//     [[]],
//     [() => {}],
//     [undefined],
//     [null],
//     [NaN],
//   ]).test("passes when not given true: %s", (given) => {
//     expect(given).not.toBeTrue();
//   });

//   test("fails when given true", () => {
//     expect(() => expect(true).not.toBeTrue()).toThrowErrorMatchingSnapshot();
//   });
// });
