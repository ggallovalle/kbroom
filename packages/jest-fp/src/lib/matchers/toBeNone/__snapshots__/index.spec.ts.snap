// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toBeNone should failed when given something that isn't none 1`] = `
" [2mexpect([22m[31mreceived[39m[2m).toBeNone()[22m

    Expected value to be none:
      [32m{\\"_tag\\": \\"None\\"}[39m
    Received:
      [31mtrue[39m"
`;
